beater/pub.go

```diff
@@ -52,13 +62,27 @@ func newPublisher(pipeline beat.Pipeline, N int, shutdownTimeout time.Duration)
 
 		// Set channel size to N - 1. One request will be actively processed by the
 		// worker, while the other concurrent requests will be buffered in the queue.
-		events: make(chan []beat.Event, N-1),
+		events: make(chan event, N-1),
+	}
+
+	for i := 0; i < numCPUToUse(); i++ {
+		go p.run()
 	}
 
-	go p.run()
 	return p, nil
 }
 
+func numCPUToUse() int {
+	maxCPU := sysinfo.Go().MaxProcs
+	numCPU := runtime.NumCPU()
```

Comment:
why just `runtime.NumCPU()` is not enough?

Reply:
Because in the `apm-server.yml` you can configure a `max_procs` with description: Sets the maximum number of CPUs that can be executing simultaneously.

If this is set then we should also respect if for the amount of consuming go routines.

Reply:
right. in that case, shouldn't `runtime.GOMAXPROCS(0)` be enough?

Reply:
I don't want to set the variable here, as this is already done when starting the apm-server in the beats code, but only want to get the number of configured CPUs.
`sysinfo.Go().MaxProcs` returns exactly what has been set when starting APM-Server. In case this hasn't been set or is an invalid number the default `runtime.NumCPU` is used.

Reply:
precisely. `runtime.GOMAXPROCS(0)` doesn't set anything, just gets you that number.
If user don't specify anything, GOMAXPROCS defaults to num of cpus, so i think you don't need any logic and any test.

Reply:
I checked [`sysinfo.Go().MaxProcs`](https://github.com/elastic/go-sysinfo/blob/master/system.go#L37) internas, and as it also calls `runtime.GOMAXPROCS(0)` I am fine with changing it.

I kept the tests, as I still see value in ensuring we create the expected amount of go routines for the transformation and publishing.